models/project_issues.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+)
+
+// ProjectIssues saves relation from issue to a project
+type ProjectIssues struct {
+	ID        int64 `xorm:"pk autoincr"`
+	IssueID   int64 `xorm:"INDEX"`
+	ProjectID int64 `xorm:"INDEX"`
+
+	// If 0, then it has not been added to a specific board in the project
+	ProjectBoardID int64 `xorm:"INDEX"`
```

Comment:
My suspicion is that this index will be quite inefficient - @guillep2k might be able to help more.

Code which relies on this index should probably search by ProjectID and AND the projectboardID test instead.

Reply:
should I remove the INDEx from ProjectBoardID?

Reply:
Why we need a `ProjectID` column since we always load issues via `ProjectBoardID`?

Reply:
For deletion?

Reply:
`issue_id` should be unique, also `issue_id` and `project_board_id` should be a composite unique.